src/hardware/pslab/api.c

```diff
@@ -18,117 +18,155 @@
  */
 
 #include <config.h>
+#include <math.h>
 #include "protocol.h"
 
-static struct sr_dev_driver pslab_driver_info;
-
-static GSList *scan(struct sr_dev_driver *di, GSList *options)
-{
-	struct drv_context *drvc;
-	GSList *devices;
-
-	(void)options;
+static const struct analog_channel analog_channels[] = {
+        {"CH1", 3},
+        {"CH2", 0},
+        {"CH3", 1},
+        {"MIC", 2},
+        {"AN4", 4},
+        {"RES", 7},
+        {"CAP", 5},
+        {"VOL", 8},
+};
 
-	devices = NULL;
-	drvc = di->context;
-	drvc->instances = NULL;
 
-	/* TODO: scan for devices, either based on a SR_CONF_CONN option
-	 * or on a USB scan. */
+static struct sr_dev_driver pslab_driver_info;
 
-	return devices;
+static GSList *scan(struct sr_dev_driver *di, GSList *options) {
+    (void) options;
+    GSList *l, *devices;
+
+    GSList *device_paths_v5 = sr_serial_find_usb(0x04D8, 0x00DF);
+
+    GSList *device_paths_v6 = sr_serial_find_usb(0x10C4, 0xEA60);
+
+    GSList *device_paths = g_slist_concat(device_paths_v5, device_paths_v6);
+
+    devices = NULL;
+    struct sr_serial_dev_inst *serial;
+    struct sr_dev_inst *sdi;
+
+    for (l = device_paths; l; l = l->next) {
+
+        serial = sr_serial_dev_inst_new(l->data, NULL);
+
+        sdi = g_new0(struct sr_dev_inst, 1);
+        sdi->status = SR_ST_INACTIVE;
+        sdi->inst_type = SR_INST_SERIAL;
+        sdi->vendor = g_strdup("PSLab");
```

Comment:
I think this should be "FOSSASIA" rather than "PSLab".

Reply:
Made the change.